src/Bicep.LangServer/Handlers/BicepDefinitionHandler.cs

```diff
@@ -45,47 +45,102 @@ public class BicepDefinitionHandler : DefinitionHandlerBase
 
         private readonly IModuleDispatcher moduleDispatcher;
 
+        private readonly IParamsCompilationManager paramsCompilationManager;
+
         public BicepDefinitionHandler(
             ISymbolResolver symbolResolver,
             ICompilationManager compilationManager,
             IFileResolver fileResolver,
             IWorkspace workspace,
             ILanguageServerFacade languageServer,
-            IModuleDispatcher moduleDispatcher) : base()
+            IModuleDispatcher moduleDispatcher,
+            IParamsCompilationManager paramsCompilationManager) : base()
         {
             this.symbolResolver = symbolResolver;
             this.compilationManager = compilationManager;
             this.fileResolver = fileResolver;
             this.workspace = workspace;
             this.languageServer = languageServer;
             this.moduleDispatcher = moduleDispatcher;
+            this.paramsCompilationManager = paramsCompilationManager;
         }
 
         public override Task<LocationOrLocationLinks> Handle(DefinitionParams request, CancellationToken cancellationToken)
         {
-            var context = this.compilationManager.GetCompilation(request.TextDocument.Uri);
-            if (context is null)
+            if (PathHelper.HasBicepparamsExension(request.TextDocument.Uri.ToUri()))
             {
-                return Task.FromResult(new LocationOrLocationLinks());
-            }
+                var paramsContext = this.paramsCompilationManager.GetCompilation(request.TextDocument.Uri);
+                if (paramsContext is null)
+                {
+                    return Task.FromResult(new LocationOrLocationLinks());
+                }
+
+                var result = this.symbolResolver.ResolveParamsSymbol(request.TextDocument.Uri, request.Position);
+                var bicepCompilation = paramsContext.ParamsSemanticModel.BicepCompilation;
+                
+                if (bicepCompilation is null)
+                {
+                    return Task.FromResult(new LocationOrLocationLinks());
+                }
+                
+                var paramsSemanticModel = paramsContext.ParamsSemanticModel;
+                var bicepSemanticModel = bicepCompilation.GetEntrypointSemanticModel();
+
+                var parameterDeclarations = bicepSemanticModel.Root.Syntax.Children.OfType<ParameterDeclarationSyntax>();
+                var parameterDeclarationSyntax = parameterDeclarations
+                    .Where(x => LanguageConstants.IdentifierComparer.Equals(bicepSemanticModel.Binder.GetSymbolInfo(x)?.Name, result?.Symbol.Name))
+                    .SingleOrDefault();
 
-            var result = this.symbolResolver.ResolveSymbol(request.TextDocument.Uri, request.Position);
+                var parameterAssignments = paramsSemanticModel.Root.Syntax.Children.OfType<ParameterAssignmentSyntax>();
```

Comment:
I think you should be able to use the symbol resolution result to get straight to the syntax in O(n) rather than having to search through the syntax tree:
```csharp
if (result.Symbol is BindableSymbol bindableSymbol &&
     bindableSymbol.NameSyntax is {} nameSyntax)
{
    ...
// use nameSyntax in place of parameterAssignmentSyntax.Name below
```